SimPEG/utils/model_utils.py

```diff
@@ -195,3 +200,128 @@ def depth_weighting(
         wz = wz[active_cells]
 
     return wz / np.nanmax(wz)
+
+
+def distance_weighting(
+    mesh: discretize.base.BaseMesh,
+    reference_locs: np.ndarray,
+    active_cells: np.ndarray | None = None,
+    exponent: float = 2.0,
+    threshold: float | None = None,
+    engine: Literal["numpy", "numba"] = "numba",
```

Comment:
is there any case (other than cases where both are trivially fast) where numpy performance is expected to be better?

My preference is to pick one or the other, rather than maintain two methods of computing the same thing

Reply:
with the switch to `cdist`, this offers the opportunity to use `cdist` options for larger range of possibilities and experimentation (at the cost of efficiency with numba)

Reply:
why would I want to use anything besides Euclidean distance for this?